---
title: "Gene List to BED"
output: html_notebook
---

```{r}
coord <- read.table("genes_mm39.txt",header = T,sep = "\t")

gl.WTup <- read.csv("table_Downreg_&_WT_specific_genes.csv")[,2]
gl.ns <- read.csv("table_NoSigChanged_genes.csv")[,2]
gl.DKOup <- read.csv("table_Upreg_&_DKO_specific_genes.csv")[,2]
```

```{r}
match<-coord[coord$Gene.stable.ID %in% gl.DKOup,]

bed.DKOspec <- data.frame( chr=paste0("chr",match$Chromosome.scaffold.name), 
                           start=match$Transcript.start..bp., 
                           end=match$Transcript.end..bp., 
                           name=match$Gene.name, 
                           score=0, 
                           strand=gsub("1","+",gsub("-1","-",match$Strand)))
                        
write.table(bed.DKOspec, file="mRNA_DKOup.bed",col.names = F,row.names = F, quote = F, sep = "\t")
```

```{r}
match<-coord[coord$Gene.stable.ID %in% gl.ns,]

bed.ns <- data.frame( chr=paste0("chr",match$Chromosome.scaffold.name), 
                           start=match$Transcript.start..bp., 
                           end=match$Transcript.end..bp., 
                           name=match$Gene.name, 
                           score=0, 
                           strand=gsub("1","+",gsub("-1","-",match$Strand)))
                        
write.table(bed.ns, file="mRNA_ns.bed",col.names = F,row.names = F, quote = F, sep = "\t")
```

```{r}

match<-coord[coord$Gene.stable.ID %in% gl.WTup,]

bed.WTspec <- data.frame( chr=paste0("chr",match$Chromosome.scaffold.name), 
                           start=match$Transcript.start..bp., 
                           end=match$Transcript.end..bp., 
                           name=match$Gene.name, 
                           score=0, 
                           strand=gsub("1","+",gsub("-1","-",match$Strand)))
                        
write.table(bed.WTspec, file="mRNA_WTup.bed",col.names = F,row.names = F, quote = F, sep = "\t")
```