---
title: "G4 CUT&Tag analysis mESC (WT, FANCJ KO, DHX36 KO, DKO)"
output: html_notebook
---

Simon Els√§sser, Karolinska Institutet (2023)

### Plots of differentially expressed genes from RNA-seq


```{r fig.width=6, fig.height=3}
bw_dir <- "/Volumes/DATA/DATA/Puck/bigwig/"
chromhmm <- "../genome/ESC_10_segments.mm39.bed"

library("wigglescout")
library("ggpubr")
library("ggplot2")
library("DESeq2")
library("dplyr")
library("ggrastr")

clean <- function (fn) {
  fn <- gsub(pattern = ".+/", "", x = fn)
  fn <- gsub(pattern = ".mm9.+", "", x = fn)
  fn <- gsub(pattern = ".mm39.+", "", x = fn)
  fn <- gsub(pattern = "_S.+", "", x = fn)
  fn <- gsub(pattern = ".scaled.bw", "", x = fn)
  fn <- gsub(pattern = ".unscaled.bw", "", x = fn)
  fn <- gsub(pattern = "_batch2", "", x = fn)
  fn <- gsub(pattern = "-", " ", x = fn)
  fn <- gsub(pattern = "_", " ", x = fn)
  fn <- gsub(pattern = " HA ", " ", x = fn)
  fn <- gsub(pattern = "D1D6", "FANCJ-/-", x = fn)
  fn <- gsub(pattern = "P2D2", "DHX36-/-", x = fn)
  fn <- gsub(pattern = "P3D4", "FANCJ-/-DHX36-/-", x = fn)
  return(fn)
}

BWs <- paste0(bw_dir,list.files(bw_dir,pattern="G4_.+_combined.bw"))
Rloop <- paste0(bw_dir,list.files(bw_dir,pattern="Rloop_.+_combined.bw"))

mypal <-c("cornflowerblue","orange","red2","#505050")
mypal3 <-c("cornflowerblue","cornflowerblue","cornflowerblue","orange","orange","orange","red2","red","red2","black","black","black")
mypal2 <-c("cornflowerblue","cornflowerblue","orange","orange","red2","red","black","black")
```

```{r}
library(bedscout)

bed_peaks_table <- read.table("G4_CnT_combined_peaks_DESeq_annotated_v2.txt", header = T, sep = "\t")
bed_peaks_table$class <- "undef"
bed_peaks_table$class[bed_peaks_table$DKO] <- "DJ"
bed_peaks_table$class[!bed_peaks_table$DKO & !bed_peaks_table$DHX36 & !bed_peaks_table$FANCJ] <- "nonDJ"

bed_peaks_bedcol <- bed_peaks_table[,c(1,2,3,16,8,5)]
colnames(bed_peaks_bedcol)[4] <- "name"
bed_peaks_bedcol$name<-as.character(bed_peaks_bedcol$name)

bed_peaks <- makeGRangesFromDataFrame(bed_peaks_bedcol,keep.extra.columns = TRUE)
mRNA <- rtracklayer::import("../genelists/mRNA_categories.tss.bed")
lncRNA <- rtracklayer::import("../genelists/lncRNA_categories.tss.bed")

mRNA_peak_ol <-  as.data.frame(impute_feature(mRNA, bed_peaks, "name"))
mRNA_peak_ol$feature[is.na(mRNA_peak_ol$feature)] <- "nopeak"

table(mRNA_peak_ol[,c(6,8)])

lncRNA_peak_ol <-  as.data.frame(impute_feature(lncRNA, bed_peaks, "name"))
lncRNA_peak_ol$feature[is.na(lncRNA_peak_ol$feature)] <- "nopeak"

table(lncRNA_peak_ol[,c(6,8)])
```

```{r}
peak_mRNA_ol <-  as.data.frame(impute_feature(bed_peaks, lncRNA, "name"))
peak_mRNA_ol$feature[is.na(peak_mRNA_ol$feature)] <- "none"
table(peak_mRNA_ol[,c(6,8)])
```

```{r fig.width=4, fig.height=3, error=F,echo=F,prompt=F,warning=F}
library(wigglescout)
library(reshape2)
mRNA_tss_G4 <- as.data.frame(bw_loci(bwfiles = BWs,loci = "../genelists/lncRNA_categories.tss.bed"))
colnames(mRNA_tss_G4)[6:9] <- c("FANCJ","DHX36","DKO","WT")

mdf <- melt(data.frame(name=rownames(mRNA_tss_G4),mRNA_tss_G4[,6:10]))

#ggviolin(mdf, x="name.1",y="value",fill="variable",palette = mypal, add="mean_sd") 
my_comparisons <- list( c("FANCJ", "WT"), c("DHX36", "WT"), c("DKO", "WT") )
#ggboxplot(mdf, x="variable",y="value",fill="variable",palette = mypal, facet.by = "name.1") + stat_compare_means(method = "t.test", comparison=my_comparisons)
ggviolin(mdf, x="variable",y="value",fill="variable",palette = mypal, facet.by = "name.1", add = "mean_sd") + stat_compare_means(method = "t.test", comparison=my_comparisons)
```

```{r fig.width=4, fig.height=3, error=F,echo=F,prompt=F,warning=F}
mRNA_tss_G4 <- as.data.frame(bw_loci(bwfiles = BWs,loci = "../genelists/mRNA_categories.tss.bed"))
colnames(mRNA_tss_G4)[6:9] <- c("FANCJ","DHX36","DKO","WT")

mdf <- melt(data.frame(name=rownames(mRNA_tss_G4),mRNA_tss_G4[,6:10]))

#ggviolin(mdf, x="name.1",y="value",fill="variable",palette = mypal, add="mean_sd") 
my_comparisons <- list( c("FANCJ", "WT"), c("DHX36", "WT"), c("DKO", "WT") )
#ggboxplot(mdf, x="variable",y="value",fill="variable",palette = mypal, facet.by = "name.1") + stat_compare_means(method = "t.test", comparison=my_comparisons)
ggviolin(mdf, x="variable",y="value",fill="variable",palette = mypal, facet.by = "name.1", add = "mean_sd") + stat_compare_means(method = "t.test", comparison=my_comparisons)
```
